csharp/src/Ice/Communicator-ObjectAdapterFactory.cs

```diff
@@ -232,14 +232,32 @@ internal void RemoveObjectAdapter(ObjectAdapter adapter)
             }
         }
 
+        // Creates and adds a new nameless ObjectAdapter
         private ObjectAdapter AddObjectAdapter(
-            string? name = null,
-            bool serializeDispatch = false,
-            TaskScheduler? taskScheduler = null,
-            IRouterPrx? router = null,
-            Protocol protocol = Protocol.Ice2)
+            bool serializeDispatch,
+            TaskScheduler? taskScheduler,
+            Protocol protocol)
         {
-            if (name != null && name.Length == 0)
+            lock (_mutex)
+            {
+                if (IsDisposed)
+                {
+                    throw new CommunicatorDisposedException();
+                }
+                var adapter = new ObjectAdapter(this, serializeDispatch, taskScheduler, protocol);
+                _adapters.Add(adapter);
+                return adapter;
+            }
+        }
+
+        // Creates and adds a new ObjectAdapter
+        private ObjectAdapter AddObjectAdapter(
```

Comment:
I think this private method should be renamed CreateObjectAdapter, with a non-defaulted router parameter to avoid ambiguity with the "main" CreateObjectAdapter.